bpf/lbmap/lbmap.go

```diff
@@ -98,7 +125,12 @@ func UpdateService(key ServiceKey, value ServiceValue) error {
 }
 
 func DeleteService(key ServiceKey) error {
-	return key.Map().Delete(key.Convert())
+	err := key.Map().Delete(key.Convert())
+	if err == nil {
+		return LookupAndDeleteServiceWeights(key)
```

Comment:
I am sorry I do not follow. I was deleting in the RR map only if the delete in main map succeeds so they are always consistent. Do you instead want me to delete in the RR map even if the delete in the main map fails ?